main.py

```diff
@@ -252,9 +252,20 @@ def post(self):
 
 class ChangePasswordHandler(BaseHandler):
     def post(self):
+        id = self.check_id() #TODO check to make sure that the account is associated with the proper id
+        if id is None:
+            self.error(403)
+            return
+        elif id == "9999": #If it is the demo account
+            self.response.write(json.dumps({"error":"You are a terrible person."}))
+            return
         email = self.request.get('email')
         old_password = self.request.get('oldpassword')
         new_password = self.request.get('newpassword')
+        if not self.check_email_id(email, id):
```

Comment:
Let's just get the email for the id here, and ignore the supplied email altogether. Safer!